widget/list.go

```diff
@@ -239,6 +254,39 @@ func (l *List) ScrollToTop() {
 	l.Refresh()
 }
 
+// TypedKey is called if a key event happens while this List is focused.
+//
+// Implements: fyne.Focusable
+func (l *List) TypedKey(event *fyne.KeyEvent) {
+	switch event.Name {
+	case fyne.KeySpace:
+		l.Select(l.currentFocus)
+	case fyne.KeyDown:
+		if f := l.Length; f != nil && l.currentFocus >= f()-1 {
```

Comment:
Can we add public APIs to get the ListItemID of the first and last visible rows? (ie a public API similar to `.currentFocus` but just getting viewport info) - that way if someone needs to handle the focus themselves by overriding these Focusable methods they have all the public APIs needed to scroll the list if necessary

and same goes for Table, Tree, and GridWrap as well

Reply:
Oh, I misread the implementation first time around and thought it was calling scrollTo only if the next-to-focus item was outside of the viewport. I guess these APIs aren't needed then but it still might be nice to have

Reply:
I'm not sure I quite follow - this isn't tabbing through items inside a list any more, the list is focused and up/down move the focused item. Tabbing moves focus to the next item in chain - either the next sibling widget or any Focusable items inside the visible list widget items.
Does that help?